crates/oxc_linter/src/rules/eslint/yoda.rs

```diff
@@ -266,61 +266,53 @@ fn is_not_yoda(expr: &BinaryExpression) -> bool {
 #[allow(clippy::cast_possible_truncation)]
 fn do_diagnostic_with_fix(expr: &BinaryExpression, ctx: &LintContext, never: bool) {
     ctx.diagnostic_with_fix(yoda_diagnostic(expr.span, never, expr.operator.as_str()), |fix| {
-        let flipped_operator = flip_operator(expr.operator);
-
-        let left_str = ctx.source_range(expr.left.span());
-        let right_str = ctx.source_range(expr.right.span());
-        let flipped_operator_str = flipped_operator.as_str();
+        let left_span = expr.left.span();
+        let right_span = expr.right.span();
 
         let operator_str = expr.operator.as_str();
-        let source_str = ctx.source_range(
-            Span::new(expr.left.span().end, expr.right.span().start)
+        let str_between_left_and_right = ctx.source_range(
+            Span::new(left_span.end, right_span.start)
         );
 
-        let source_chars = source_str.char_indices().collect::<Vec<_>>();
-
-        let search_start_position = expr.left.span().end;
-
-        let operator_position_start = source_chars.windows(operator_str.len()).find(|str|  {
-            if str.iter().enumerate().all(|(i, (_pos, c))| *c == operator_str.chars().nth(i).unwrap()) {
-                !ctx.comments().iter().any(|c| {
-                    c.span.start <= (str[0].0 as u32) + search_start_position && (str[0].0 as u32) + operator_str.len() as u32 + search_start_position <= c.span.end
-                })
-            } else {
-                false
-            }
-        });
-
-        let Some(operator_position_start) = operator_position_start else {
-            debug_assert!(false);
-            return fix.noop();
-        };
-
-        let operator_position_start = search_start_position + operator_position_start[0].0 as u32;
+        let (operator_start, operator_end) = str_between_left_and_right
+            .as_bytes()
```

Comment:
is there any risk of this causing issues with multi byte chars?
multi byte chars are almost always the cause of our panics 🫠

Reply:
I don't think so. `operator_str` is always ASCII, so if `chunk == operator_str.as_bytes()` then `pos_start` and `pos_end` must be on UTF8 character boundaries (as they're the start and end of `operator_str`.

Multibyte unicode characters never contain bytes in the ASCII range (0-127). So e.g. `=` can't appear in the middle of a multi-byte sequence.

Reply:
Incidentally, same goes for multi-byte characters. No 2 bytes of a 3-byte character can be a valid 2-byte character. So there's never any ambiguity.